external-crates/move/crates/move-unit-test/src/test_runner.rs

```diff
@@ -289,105 +303,219 @@ impl SharedTestingConfig {
         let mut stats = TestStatistics::new();
 
         for (function_name, test_info) in &test_plan.tests {
-            let (_cs_result, _ext_result, exec_result, test_run_info) =
-                self.execute_via_move_vm(test_plan, function_name, test_info);
-
-            match exec_result {
-                Err(err) => {
-                    let sub_status = err.sub_status().and_then(|status| {
-                        convert_clever_move_abort_error(status, err.location(), global_test_context)
+            let arguments = if test_info
+                .arguments
+                .iter()
+                .all(|arg| matches!(arg, TestArgument::Value(_)))
+            {
+                vec![(
+                    None,
+                    test_info
+                        .arguments
+                        .iter()
+                        .map(|arg| match arg {
+                            TestArgument::Value(v) => v.clone(),
+                            _ => unreachable!(),
+                        })
+                        .collect::<Vec<_>>(),
+                )]
+            } else {
+                let mut args = vec![];
+                for i in 0..self.num_iters {
+                    let mut iter_args = vec![];
+                    let prng_seed = self.prng_seed.unwrap_or(if self.deterministic_generation {
```

Comment:
unwrap_or_else? maybe? 
What does clippy say?

Reply:
Clippy is happy with this :) But `unwrap_or_else` might be slightly nicer.